lib/cuffsert/rxcfclient.rb

```diff
@@ -94,8 +99,15 @@ def record_start_time
     end
 
     def stack_finished?(stack_id, event)
-      event[:physical_resource_id] == stack_id &&
-        FINAL_STATES.include?(event[:resource_status])
+      return false unless event[:physical_resource_id] == stack_id
+      case CuffSert.state_category(event[:resource_status])
+      when :bad
+        raise RxCFError, "Stack #{event.logical_resource_id} finished in state #{event.resource_status}"
```

Comment:
Not sure how this all fits together, but won't raising here mean that we risk not processing additional events?

Reply:
That's true, good catch. Concretely, cleanup-events will show up after the stack event. Not sure if you can get cleanup-events after a failure, but let's not take that chance.

Reply:
Fixed.